index.bs

```diff
@@ -492,22 +504,30 @@ To <dfn>trigger attribution</dfn> given an [=attribution trigger=] |trigger| run
 1. Set |matchingSources| to the result of [=list/sort in descending order|sorting=] |matchingSources|
     in descending order, with |a| being less than |b| if any of the following are true:
       * |a|'s [=attribution source/priority=] is less than |b|'s [=attribution source/priority=].
-      * |a|'s [=attribution source/priority=] is equal to |b|'s [=attribution source/priority=] and |a|'s 
+      * |a|'s [=attribution source/priority=] is equal to |b|'s [=attribution source/priority=] and |a|'s
          [=attribution source/source time=] is less than |b|'s [=attribution source/source time=].
 1. Let |sourceToAttribute| be the first item in |matchingSources|.
 1. If |trigger|'s [=attribution trigger/dedup key=] is not null and |sourceToAttribute|'s
     [=attribution source/dedup keys=] [=list/contains=] it, return.
+1. Let |report| be the result of running [=obtain a report=] with |sourceToAttribute| and |trigger|.
+1. If |sourceToAttribute|'s [=attribution source/number of reports=] value is greater than or equal to the
+    user agent's [=max reports per source=] value, then:
+    1. Let |matchingReports| be all entries in the [=attribution report cache=] where all of the following are true:
+         * entry's [=attribution report/report time=] and |report|'s [=attribution report/report time=] are equal.
+         * entry's [=attribution report/source=] and |report|'s [=attribution report/source=] are the same.
+    1. Set |matchingReports| to the result of [=list/sort in ascending order|sorting=] |matchingReports|
+        in ascending order, with |a| being less than |b| if |a|'s [=attribution report/trigger priority=] is less than |b|'s [=attribution report/trigger priority=].
+    1. Let |lowestPriorityReport| be the first item in |matchingReports|.
+    1. If |report|'s [=attribution report/trigger priority=] is less than or equal to |lowestPriorityReport|'s [=attribution report/trigger priority=], return.
+    1. Remove |lowestPriorityReport| from the [=attribution report cache=].
 1. [=list/Remove=] |sourceToAttribute| from |matchingSources|.
 1. For each |item| of |matchingSources|:
     1. [=list/Remove=] |item| from the [=attribution source cache=].
-1. Let |report| be the result of running [=obtain a report=] with |sourceToAttribute| and |trigger|.
 1. If the [=list/size=] of the [=attribution report cache=] is greater than an implementation-defined limit, return.
 1. Add |report| to the [=attribution report cache=].
 1. Increment |sourceToAttribute|'s [=attribution source/number of reports=] value by 1.
 1. If |trigger|'s [=attribution trigger/dedup key=] is not null, [=list/append=] it to |sourceToAttribute|'s
     [=attribution source/dedup keys=].
-1. If |sourceToAttribute|'s [=attribution source/number of reports=] value is greater than or equal to the
-    user agent's [=max reports per source=] value, [=list/remove=] |sourceToAttribute| from the [=attribution source cache=].
```

Comment:
I'm not sure whether this should be incorporated into the replacement section above.

Reply:
I think it should be:

If matchingReports is empty we cannot do prioritization and should delete the source